megamek/src/megamek/common/MiscType.java

```diff
@@ -864,30 +809,19 @@ public double getTonnage(Entity entity, int location) {
                 }
             }
             if (entity.hasETypeFlag(Entity.ETYPE_JUMPSHIP)) {
-                return Math.ceil(entity.getWeight() * pct);
+                return RoundWeight.nextTon(entity.getWeight() * pct);
             } else {
-                return Math.ceil((entity.getWeight() * pct) * 2.0) / 2.0;
+                return defaultRounding.round(entity.getWeight() * pct, entity);
             }
         } else if (hasFlag(MiscType.F_ATAC)) {
             //TODO Neo - pg IO 146 Each drone that it can control adds 150 ton to weight.
-            double tWeight = 0;
-            tWeight = Math.ceil((entity.getWeight() * 0.02) * 2) / 2.0;
-            if (tWeight >50000) {
-                return 50000;
-            } else {
-                return Math.ceil((entity.getWeight() * 0.02) * 2) / 2.0;
-            }
+            double tWeight = defaultRounding.round(entity.getWeight() * 0.02, entity);
+            return Math.min(tWeight, 50000);
         } else if (hasFlag(MiscType.F_DTAC)) {
             //TODO Neo - pg IO 146 Each drone that it can control adds 150 ton to weight.
-            return Math.ceil((entity.getWeight() * 0.03) * 2) / 2.0;
+            return defaultRounding.round(entity.getWeight() * 0.03, entity);
         } else if (hasFlag(MiscType.F_SDS_DESTRUCT)) {
-            double tWeight = 0;
-            tWeight = Math.ceil((entity.getWeight() * 0.1) * 2) / 2.0;
-            if (tWeight >10000) {
-                return 10000;
-            } else {
-                return Math.ceil((entity.getWeight() * 0.02) * 2) / 2.0;
-            }
+            return Math.min(RoundWeight.nextTon(entity.getWeight() * 0.1), 10000);
```

Comment:
The old code looks like to the next half ton (now `defaultRounding.round(...)` I believe)

Reply:
This is Okay, SDS Self Destruct had the wrong formula and was corrected.